src/Core/src/Handlers/Button/ButtonHandler.iOS.cs

```diff
@@ -200,12 +202,50 @@ partial class ButtonImageSourcePartSetter
 			public override void SetImageSource(UIImage? platformImage)
 			{
 				if (Handler?.PlatformView is not UIButton button)
-					return;
+ 					return;
+
+ 				nfloat buttonSize = 0;
+
+ #pragma warning disable CA1422 // Validate platform compatibility
+ 				var contentEdgeInsets = button.ContentEdgeInsets;
+ #pragma warning restore CA1422 // Validate platform compatibility
+
+ 				if (button.Bounds != CGRect.Empty)
+ 				{
+ 					if (button.Bounds.Height > button.Bounds.Width)
+ 						buttonSize = button.Bounds.Width - (contentEdgeInsets.Left + contentEdgeInsets.Right);
+ 					else
+ 						buttonSize = button.Bounds.Height - (contentEdgeInsets.Top + contentEdgeInsets.Bottom);
+ 				}
+
+ 				try
+ 				{
+ 					if (buttonSize != 0)
+ 						platformImage = ResizeImageSource(platformImage, buttonSize, buttonSize);
+
+ 					platformImage = platformImage?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+
+ 					button.SetImage(platformImage, UIControlState.Normal);
+ 				}
+ 				catch (Exception)
+ 				{
+ 					Handler.MauiContext?.CreateLogger<ButtonHandler>()?.LogWarning("Can not load Button ImageSource");
+ 				}
+			}
+			
+			static UIImage? ResizeImageSource(UIImage? sourceImage, nfloat maxWidth, nfloat maxHeight)
```

Comment:
The fix is the same applied on https://github.com/dotnet/maui/pull/20953 from https://github.com/dotnet/maui/pull/19834

We adapt the size of the Button image. For this reason, once PR #20953 is merged, this PR will only include a specific manual test for Catalyst, UIDeviceFamily 6.